experiments/conductor/cmd/runner/controller_commands.go

```diff
@@ -0,0 +1,434 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runner
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// generateControllerClient creates a controller client for the branch
+// This implements the logic from 01-create-controller-client.sh
+func generateControllerClient(ctx context.Context, opts *RunnerOptions, branch Branch) ([]string, error) {
+	log.Printf("Generating controller client for branch %s", branch.Name)
+
+	// Check if we have the required fields
+	if branch.ProtoSvc == "" {
```

Comment:
Can we change `ProtoSvc` to some other name to align with other steps?

Reply:
change to what ? Also this would mean updating metadata for all. We can discuss in standup if this is important.